src/app/util/im-client-callbacks.cpp

```diff
@@ -437,8 +437,8 @@ bool IMSubscribeResponseCallback(const chip::app::ReadClient * apSubscribeClient
     CHIP_ERROR err                           = CHIP_NO_ERROR;
     Callback::Cancelable * onSuccessCallback = nullptr;
     Callback::Cancelable * onFailureCallback = nullptr;
-    err = gCallbacks.GetResponseCallback(apSubscribeClient->GetExchangeContext()->GetSecureSession().GetPeerNodeId(),
-                                         sequenceNumber, &onSuccessCallback, &onFailureCallback);
+    err =
+        gCallbacks.GetResponseCallback(apSubscribeClient->GetPeerNodeId(), sequenceNumber, &onSuccessCallback, &onFailureCallback);
```

Comment:
This is insufficient to identify a peer, since fabric information is needed as well. Since this PR isn't altering the GetResponseCallback API, I think it needs a follow-up.

Reply:
I have seen this exact code path happily cross its wires across fabrics and route callbacks to the wrong context.

Indeed, this is a really big security problem.